dhcpv6/option_vendor_opts.go

```diff
@@ -0,0 +1,161 @@
+      .                                                               .
+      .                   option-data (sub-options)         					.
+      .                                                               .
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+				Sub-Option
+	 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	|          opt-code             |             option-len        |
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	.                                                               .
+	.                          option-data                          .
+	.                                                               .
+	+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+*/
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+)
+
+// OptVendorOpts represents a DHCPv6 Status Code option
+type OptVendorOpts struct {
+	EnterpriseNumber uint32
+	VendorOpts       []Option
+}
+
+// Code returns the option code
+func (op *OptVendorOpts) Code() OptionCode {
+	return OptionVendorOpts
+}
+
+// ToBytes serializes the option and returns it as a sequence of bytes
+func (op *OptVendorOpts) ToBytes() []byte {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint16(buf[0:2], uint16(OptionVendorOpts))
+	binary.BigEndian.PutUint16(buf[2:4], uint16(op.Length()))
+	binary.BigEndian.PutUint32(buf[4:8], uint32(op.EnterpriseNumber))
+	for _, opt := range op.VendorOpts {
+		buf = append(buf, opt.ToBytes()...)
+	}
+	return buf
+}
+
+// Length returns the option length
+func (op *OptVendorOpts) Length() int {
+	l := 4 // 4 bytes for Enterprise Number
+	for _, opt := range op.VendorOpts {
+		l += 4 + opt.Length() // 4 bytes for Code and Length from Vendor
+	}
+	return l
+}
+
+// String returns a string representation of the VendorOpts data
+func (op *OptVendorOpts) String() string {
+	return fmt.Sprintf("OptVendorOpts{enterprisenum=%v, vendorOpts=%v}",
+		op.EnterpriseNumber, op.VendorOpts,
+	)
+}
+
+// ParseOptVendorOpts builds an OptVendorOpts structure from a sequence of bytes.
+// The input data does not include option code and length bytes.
+func ParseOptVendorOpts(data []byte) (*OptVendorOpts, error) {
+	opt := OptVendorOpts{}
+	if len(data) < 4 {
+		return nil, fmt.Errorf("Invalid vendor opts data length. Expected at least 4 bytes, got %v", len(data))
+	}
+	opt.EnterpriseNumber = binary.BigEndian.Uint32(data[:4])
+
+	var err error
+	opt.VendorOpts, err = VendorOptionsFromBytes(data[4:])
+	if err != nil {
+		return nil, err
+	}
+	return &opt, nil
+}
+
+// VendorOptionsFromBytes builds a slice of GenericOptions from a slice of bytes
+func VendorOptionsFromBytes(data []byte) ([]Option, error) {
+	// Parse a sequence of bytes until the end and build a list of options from
+	// it. Returns an error if any invalid option or length is found.
+	options := make([]Option, 0, 10)
```

Comment:
I imagine that 10 is an arbitrary capacity, maybe too high. Do you know from experience what's a common number of options in a vendor opt? We may tune around that instead, or it could be even 0 since this doesn't seem to be a performance offender